core/sdl/dreampicoport.cpp

```diff
@@ -625,498 +653,843 @@ class DreamPicoPortSerialHandler
-		serial->sendCmd(s.str(), timeout_ms);
+	void setDefaultMapping(const std::shared_ptr<InputMapping>& mapping) const override {
+		// Since this is a real DC controller, no deadzone adjustment is needed
+		mapping->dead_zone = 0.0f;
+		// Map the things not set by SDL
+		mapping->set_button(DC_BTN_C, 2);
+		mapping->set_button(DC_BTN_Z, 5);
+		mapping->set_button(DC_BTN_D, 10);
+		mapping->set_button(DC_DPAD2_UP, 9);
+		mapping->set_button(DC_DPAD2_DOWN, 8);
+		mapping->set_button(DC_DPAD2_LEFT, 7);
+		mapping->set_button(DC_DPAD2_RIGHT, 6);
 	}
-}
 
-int DreamPicoPort::hardwareBus() const {
-	return hardware_bus;
-}
+	const char *getButtonName(u32 code) const override {
+		switch (code) {
+			// Coincides with buttons setup in setDefaultMapping
+			case 2: return "C";
+			case 5: return "Z";
+			case 10: return "D";
+			case 9: return "DPad2 Up";
+			case 8: return "DPad2 Down";
+			case 7: return "DPad2 Left";
+			case 6: return "DPad2 Right";
+
+			// These buttons are normally not physically accessible but are mapped on DreamPicoPort
+			case 12: return "VMU1 A";
+			case 15: return "VMU1 B";
+			case 16: return "VMU1 Up";
+			case 17: return "VMU1 Down";
+			case 18: return "VMU1 Left";
+			case 19: return "VMU1 Right";
+
+			default: return nullptr; // no override
+		}
+	}
 
-bool DreamPicoPort::isHardwareBusImplied() const {
-	return is_hardware_bus_implied;
-}
+	std::string getUniqueId() const override {
+		return unique_id;
+	}
 
-bool DreamPicoPort::isSingleDevice() const {
-	return is_single_device;
-}
+	void changeBus(int newBus) override {
+		software_bus = newBus;
+		if (dpp_comms) {
+			dpp_comms->changeSoftwareBus(software_bus);
+		}
+	}
 
-void DreamPicoPort::determineHardwareBus(int joystick_idx, SDL_Joystick* sdl_joystick) {
-	// This function determines what bus index to use when communicating with the hardware.
+	std::string getName() const override {
+		return getName(" ");
+	}
 
-	// Set the serial number if found by SDL Joystick
-	const char* joystick_serial = SDL_JoystickGetSerial(sdl_joystick);
-	if (joystick_serial) {
-		serial_number = joystick_serial;
+	bool needsRefresh() override {
+		// TODO: implementing this method may also help to support hot plugging of VMUs/rumble packs here.
+		return false;
 	}
 
-#if defined(_WIN32)
-	// This only works in Windows because the joystick_path is not given in other OSes
-	const char* joystick_name = SDL_JoystickName(sdl_joystick);
-	const char* joystick_path = SDL_JoystickPath(sdl_joystick);
-
-	struct SDL_hid_device_info* devs = SDL_hid_enumerate(VID, PID);
-	if (devs) {
-		struct SDL_hid_device_info* my_dev = nullptr;
-
-		if (!devs->next) {
-			// Only single device found, so this is simple (host-1p firmware used)
-			hardware_bus = 0;
-			is_hardware_bus_implied = false;
-			is_single_device = true;
-			my_dev = devs;
-		} else {
-			struct SDL_hid_device_info* it = devs;
+	bool isConnected() override {
+		return !!dpp_comms;
```

Comment:
So, there's no need for this function to detect a situation like the cable being disconnected, right? Because the gamepad disconnection will be detected and its destructor will clean everything up?